gcs/conn.go

```diff
@@ -46,6 +46,15 @@ type Conn interface {
 	OpenBucket(
 		ctx context.Context,
 		name string) (b Bucket, err error)
+
+	// Return a Bucket object representing the GCS requester pays bucket with the given name.
+	// Set the project sepcified by userProject as the project to be billed for requests made.
+	// (https://cloud.google.com/storage/docs/requester-pays)
+	// Attempt to fail early in the case of bad credentials.
+	OpenRequesterPaysBucket(
```

Comment:
Instead of a new method, could you please add an OpenBucketOptions struct and make OpenBucket take that? (Please move "name" into the struct too.)

Probably want to call the field in the struct BillingProject, since this is less opaque than "user project". It appears to be the term used by the GCS documentation outside of the query parameter itself. Let me know if this doesn't make sense to you as a user.

Reply:
Yup, I'll do that.

FYI, I used "user project" because that is what both gsutil and the JSON API call it. Though, I agree that billing project is a much better name.